### ISSUE TITLE
Flip the Scaleway IP -> Server relationship to Server -> IP

### ISSUE THREAD

**Issue by tinoadams** [hashicorp/terraform#14175]
### Terraform Version
Terraform v0.9.4

### Affected Resource(s)
- scaleway_ip
- scaleway_server

### Terraform Configuration Files
```hcl
variable "gateway_count" {
  default = 2
}
resource "scaleway_ip" "public_ip" {
  count = "${var.gateway_count}"
  server = "${element(scaleway_server.gateway.*.id, count.index)}"
}
resource "scaleway_server" "gateway" {
  count = "${var.gateway_count}"
  name  = "gateway_${count.index}"
  image = "89457135-d446-41ba-a8df-d53e5bb54710"
  type  = "C2S"
  connection {
    type         = "ssh"
    user         = "root"
    agent        = true
  }
  provisioner "remote-exec" {
    inline = [
      "echo 'SUCCESS'",
    ]
  }
}
```

### Expected Behavior
The public IP should be created, then the server, then the IP attached to the server and finally the provisioner executed.

Flipping the relationship direction to Server -> IP also means I could potentially provide the server with an existing IP (not creating a resource via Terraform). Quite useful as Scaleway IP's are reserved and can be re-used.

### Actual Behavior
The relationship between the two resources scaleway_ip and scaleway_server is such that the server is created before the IP ie. the IP is only attached after the Server resource is completely created/provisioned.

The provisioner never manages to connect as the public IP is not attached yet.

### Steps to Reproduce
terraform apply

**Comment by nicolai86** [comment]
@tinoadams thank you for reporting this. 

I'm interested: why not set the `dynamic_ip_required` attribute to get a public IP for provisioning, and use `scaleway_ip` too to change the IP after provisioning? 

```
variable "gateway_count" {
  default = 2
}
resource "scaleway_ip" "public_ip" {
  count = "${var.gateway_count}"
  server = "${element(scaleway_server.gateway.*.id, count.index)}"
}
resource "scaleway_server" "gateway" {
  dynamic_ip_required = true # so provisioning succeeds
  count = "${var.gateway_count}"
  name  = "gateway_${count.index}"
  image = "89457135-d446-41ba-a8df-d53e5bb54710"
  type  = "C2S"
  connection {
    type         = "ssh"
    user         = "root"
    agent        = true
  }
  provisioner "remote-exec" {
    inline = [
      "echo 'SUCCESS'",
    ]
  }
}
```

**Comment by stack72** [comment]
Closed via #14515
